### PR TITLE
Add Build Instructions to README.md

### LINKED ISSUE #23: Add build instructions to the readme
No where in the README are there build dependencies or instructions. Apache is very source code focused, so instructions should first and foremost be about how to build things from source. It's fine to also provide convenience binaries, whether they be from downloads.apache.org or via the marketplace, but that's importtation information that should be in this README. It doesn't need to be part of this PR, but a new issue should be opened to add that information.

_Originally posted by @stevedlawrence in https://github.com/apache/daffodil-vscode/pull/22#discussion_r727022373_

### PR DESCRIPTION
**Pull request by Shanedell**
Changes:
- Added instructions of how to build from source to the `README.md`

To view changes:
- https://github.com/Shanedell/daffodil-vscode/tree/issue_23_build_in_readme

Closes #23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,14 +23,79 @@
...
+  ```
+
+* Package
+  ```bash
+  yarn package
+  ```
+
+  * Creates file `daffodil-debugger-${package.json_version}.vsix`
+
+* Install VSIX file
```
Comment (stevedlawrence) [inline README.md:94]: I think this belongs in an Install section.

File: README.md [hunk 2]
```diff
@@ -23,14 +23,79 @@
...
+  ```bash
+  yarn compile
+  ```
+
+* Package
+  ```bash
+  yarn package
+  ```
+
+  * Creates file `daffodil-debugger-${package.json_version}.vsix`
```
Comment (stevedlawrence) [inline README.md:92]: This is minor, but the heavy use of lists and nesting I find a bit hard to follow. I would suggest using sections and subsections rather than lists and sublists.

File: README.md [hunk 3]
```diff
@@ -23,14 +23,78 @@
...
+## Installation - via release
 
 Until the extension is available in the [VS Code Extension Marketplace](https://marketplace.visualstudio.com/vscode), please download the latest `.vsix` file from the [releases page](https://github.com/apache/daffodil-vscode/releases). Then install it by either:
   * using the "Extensions: Install from VSIX" command from within VS Code (open the Command Palette with Shift-Command-P, then type `vsix` to bring up the command and pointing it at the downloaded `vsix` file); or
   * on the command-line via `code --install-extension <path-to-downloaded-vsix>`.
 
 
+## Installation - via source
+
+:exclamation: Currently only Daffodil 3.1.0 is support, will be updated in the future :exclamation:
```
[the remark this answers is not in the document]
Comment (Shanedell) [reply README.md:40]: Probably, will delete

File: README.md [hunk 4]
```diff
@@ -23,14 +23,79 @@
 
 This is a VS Code extension which enables the interactive debugging of DFDL schema parsing using [Apache Daffodil](https://daffodil.apache.org/).
 
+## Build Requirements
+- SDK 8 or higher
+- SBT 0.13.8 or higher
+- Node 10 or higher
```
[the remark this answers is not in the document]
Comment (Shanedell) [reply README.md:29]: yeah you both are correct I will fix these issues

File: package.json [hunk 5]
```diff
@@ -31,7 +31,9 @@
     "watch2": "tsc -watch -p ./",
     "package": "vsce package",
     "publish": "vsce publish",
-    "package-ext": "webpack --mode production --config ./build/extension.webpack.config.js"
+    "package-ext": "webpack --mode production --config ./build/extension.webpack.config.js",
+    "sbt": "sbt universal:packageBin",
+    "build": "yarn sbt && yarn install && yarn compile && yarn package"
```
Comment (stevedlawrence) [inline package.json:36]: Cool!

Just a question about yarn, is this standard to have a yarn script run other yarn scripts like this? Thinking how sbt works, you just specify command dependencies. So you could specify that `yarn package` depends on `yarn compile`, which depends on `yarn install`. That way you can tell people to just run `yarn package` and if they haven't run the previous commands, they would get run because they are dependent commands.

But maybe yarn doesn't work like that, and this `yarn build` is the standard way to do this kind of thing?

Comment (Shanedell) [reply package.json:36]: From what I saw it seems the only way to do is the way I have implemented. Yarn doesn't allow you to specify other scripts a script depends on so that they will run prior.

I believe the only way it functions this way is if you have something like `build` and `prebuild` then run `yarn run build` I think it will automatically run prebuild first but not completely sure on that.

**Comment by jw3** [comment]
What is the status of this?

File: README.md [hunk 6]
```diff
@@ -21,14 +21,81 @@
...
+## License
+
+Apache Daffodil VSCode is licensed under the [Apache License, v2.0].
+
+[Apache License, v2.0]: https://www.apache.org/licenses/LICENSE-2.0
+[Apache RAT]: https://creadur.apache.org/rat/
+[Daffodil JIRA]: https://issues.apache.org/jira/projects/DAFFODIL/
+[Github Actions]: https://github.com/apache/daffodil-vscode/actions?query=branch%3Amain+
+[Releases]: https://github.com/apache/daffodil-vscode/releases
+[SBT]: https://www.scala-sbt.org/
```
Comment (stevedlawrence) [inline README.md:100]: None of these link references are used, except for the Apache license. All except that one can be removed.

File: README.md [hunk 7]
```diff
@@ -21,13 +21,65 @@
...
+* Via "Extensions: Install from VSIX" command from within VS Code.
+  * Open the Command Palette with Shift-Command-P (Mac) OR Shift-Ctrl-P (Windows/Linux)
+  * Type `vsix` to bring up the command and pointing it at the `vsix` file
 
+* Via command line
+  ```bash
+  code --install-extension daffodil-debugger-*.vsix
+  ```
 
 ## Usage
```
[the remark this answers is not in the document]
Comment (stevedlawrence) [reply README.md:81]: JIRA is used just for Daffodil. We are using the GitHub issues for the vscode extension.

**Review by stevedlawrence — APPROVED** [review]
+1 once the links are fixed up

File: README.md [hunk 8]
```diff
@@ -21,14 +21,75 @@
...
 
 Please refer to the Wiki page at https://github.com/apache/daffodil-vscode/wiki
+
+## Getting Help
+
+You can ask questions on the dev@daffodil.apache.org or
+users@daffodil.apache.org mailing lists.  You can report bugs via the
+[Daffodil JIRA].
```
Comment (jw3) [inline README.md:89]: This should be GitHub Issues not Daffodil JIRA, with link to the issues page.
